cmd/admin/cert_test.go

```diff
@@ -0,0 +1,282 @@
+package main
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io"
+	"os"
+	"path"
+	"slices"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/letsencrypt/boulder/core"
+	corepb "github.com/letsencrypt/boulder/core/proto"
+	berrors "github.com/letsencrypt/boulder/errors"
+	blog "github.com/letsencrypt/boulder/log"
+	rapb "github.com/letsencrypt/boulder/ra/proto"
+	"github.com/letsencrypt/boulder/revocation"
+	"github.com/letsencrypt/boulder/sa"
+	sapb "github.com/letsencrypt/boulder/sa/proto"
+	"github.com/letsencrypt/boulder/test"
+	"github.com/letsencrypt/boulder/test/vars"
+)
+
+// mockSAWithIncident is a mock which only implements the SerialsForIncident
+// gRPC method. It can be initialized with a set of serials for that method
+// to return.
+type mockSAWithIncident struct {
+	sapb.StorageAuthorityReadOnlyClient
+	incidentSerials []string
+}
+
+// SerialsForIncident returns a fake gRPC stream client object which itself
+// will return the mockSAWithIncident's serials in order.
```

Comment:
This "in order" I assume means "returns mockSAWithIncident's serials in the order in which they were received from the gRPC stream" ?

Reply:
It means "return's mockSAWithIncident's serials in the order in which they appear in its `incidentSerials` slice"